### DISCUSSION TITLE
Pdf `insert_text` with cyrilic encoding.

### DISCUSSION THREAD

**Discussion by vb64** [pymupdf/PyMuPDF#2687] [Q&A] [answered]
Hi there.

I ran the slightly modified example code from the documentation ([How to Use Non-Standard Encoding](https://pymupdf.readthedocs.io/en/latest/recipes-text.html#how-to-use-non-standard-encoding)) on two platforms.

- Microsoft Windows [Version 10.0.14393]
- Ubuntu 20.04.6 LTS

```python
import fitz

assert 'PyMuPDF 1.23.3:' in fitz.__doc__
doc=fitz.open()
page = doc.new_page()
shape = page.new_shape()
shape.insert_text((50,70), "Hello world!", fontname="HELV", encoding=fitz.TEXT_ENCODING_CYRILLIC)
shape.insert_text((50,90), "Привет world!", fontname="HELV", encoding=fitz.TEXT_ENCODING_CYRILLIC)
shape.insert_text((50,110), "Привет мир!", fontname="HELV", encoding=fitz.TEXT_ENCODING_CYRILLIC)
shape.commit()
doc.save("t.pdf")
```

And I got the same results.

![image](https://github.com/pymupdf/PyMuPDF/assets/8738387/dfbffa46-90d9-4714-9fdd-9c25de4eb481)

How can I get the correct text in Cyrillic in PDF?

**Comment by JorjMcKie** [comment, marked as answer]
I would avoid using the built-in Base-14 fonts. Try to use the file-based versions that also come with (Py-) MuPDF, which contain lots more of supported characters - beyond the 255 Unicode limits of Base-14.

Works like so:
```python
import fitz

assert 'PyMuPDF 1.23.3:' in fitz.__doc__
font = fitz.Font("helv")  # looks exactly equal to its Base-14 brother
doc=fitz.open()
page = doc.new_page()
page.insert_font(fontname="myhelv", fontbuffer=font.buffer)
shape = page.new_shape()
shape.insert_text((50,70), "Hello world!", fontname="myhelv")
shape.insert_text((50,90), "Привет world!", fontname="myhelv")
shape.insert_text((50,110), "Привет мир!", fontname="myhelv")
shape.commit()
doc.save("t.pdf")
```
Result:
![grafik](https://github.com/pymupdf/PyMuPDF/assets/8290722/1e556822-bda6-4f9e-86d0-f582ecbf949a)

**Comment by JorjMcKie** [comment]
The only thing to watch out for: do not use the "reserved" names in "fontname" parameters like "helv". Do something like that "myhelv" stuff.

**Comment by vb64** [comment]
Thanks, it works!

**Comment by JorjMcKie** [comment]
I think we should adapt the documentation accordingly. You were seduced to follow a path that is not recommendable.

**Reply by JorjMcKie** [reply]
Another comment about file sizing:
The file-based Base-14 equivalents unfortunately are not subsettable. Your PDF there has a 43KB size - which cannot be reduced.
If you use a font like "figo" (needs pymupdf-fonts installed) or "cjk" (standard), you will have a similar looking result (even nicer), but they are subsettable - leading to 4 KB only:
```python
import fitz

assert "PyMuPDF 1.23.3:" in fitz.__doc__
font = fitz.Font("figo")  # needs package pymupdf-fonts
doc = fitz.open()
page = doc.new_page()
page.insert_font(fontname="myhelv", fontbuffer=font.buffer)
shape = page.new_shape()
shape.insert_text((50, 70), "Hello world!", fontname="myhelv")
shape.insert_text((50, 90), "Привет world!", fontname="myhelv")
shape.insert_text((50, 110), "Привет мир!", fontname="myhelv")
shape.commit()
doc.subset_fonts(verbose=True)  # shrinks all fonts to actually used characters
doc.ez_save("t.pdf")  # save with garbage option!
```
